### PR TITLE
Use Liger Kernels for acceleration and memory reduction

### PR DESCRIPTION
**Pull request by kabachuha**
[Liger Kernel](https://github.com/linkedin/liger-kernel) is a collection of optimized Triton kernels designed by LinkedIn.

According to the project page, it can effectively increase throughput by 20% and reduce memory usage by 60%, which is essential to process high resolution and frame count videos. Unlike mechanisms like sage attention and caches, these are lossless, meaning no quality drop. Liger Kernels are already integrated with Huggingface Transformers library and can be activated with a simple keyword argument. They do not require any additional dependencies, only Torch and Triton, which come by default.

The following optimizations can be employed in Wan: RMSNorm, LayerNorm, Softmax and Rotary Positional embed. (the full list is on the project page)

---

Implementation progress:

- [x] LigerRMSNorm, LigerLayerNorm
- [x] Liger Softmax
- [ ] Liger RoPE
- [ ] Benchmarking

### CODE REVIEW

**Comment by scottmudge** [comment]
Am I right in assuming this is incompatible with torch.compile()? It was throwing errors when I tried these Liger kernels.

**Comment by kabachuha** [comment]
Yes, probably. However, the liger parts can be guarded with torch.compile.disable, not affecting the rest of the model (like they integrate liger kernels with torch compile in transformers)

**Comment by scottmudge** [comment]
> However, the liger parts can be guarded with torch.compile.disable

Ah that's a good point. I'll try guarding those areas and see if there's any speed/memory benefit over the existing implementation + torch.compile().

**Comment by kijai** [comment]
But are they faster than with torch.compile? I did look into these a bit before and I'd be most interested in the RoPE part as that's a VRAM peak bottleneck unlike the norms, the current RoPE can be torch.compiled too though.

**Comment by kabachuha** [comment]
They have Colab notebooks to benchmark individual components (RMSNorm, RoPE) against torch here https://linkedin.github.io/Liger-Kernel/Low-Level-APIs/

I plan to test them in action this weekend + add liger RoPE as an option

**Comment by Hoernchen** [comment]
@kabachuha Any results yet? I found that repo and wondered if it would be useful at all for inference, because everything just mentions training..

**Comment by kabachuha** [comment]
It gives me approx 4 seconds of speedup, which is quite small, however it's without RoPE

The MLP/Norms seems not to be the bottleneck here, more likely it's the attention, though benchmarking individual transformer's parts would be nice for that

To integrate RoPE it's needed to track the actual shapes of q, k, v, which is a bit of a puzzle to make work in a correct way because they are reshaped and the function expects slightly different shapes, but should be possible, and many repos use fused RoPE

**Comment by StrongerXi** [comment]
> It gives me approx 4 seconds of speedup

Is that e2e speed up? Or for a specific kernel?

**Comment by kabachuha** [comment]
End to end

So, with this small speedup, I think, I will close it
